content/misc/ollama-openwebui/index.md

```diff
@@ -0,0 +1,222 @@
+
+First we need to check availability of our instance type used here (g4dn.xlarge), it should be in same region of the cluster. Note you can use also Graviton based instance (ARM64) like g5g* but only on HCP 4.16+ cluster.
+
+```bash
+for region in $(aws ec2 describe-regions --query 'Regions[].RegionName' --output text); do
+    echo "Region: $region"
+    aws ec2 describe-instance-type-offerings --location-type availability-zone \
+    --filters Name=instance-type,Values=g4dn.xlarge --region $region \
+    --query 'InstanceTypeOfferings[].Location' --output table
+    echo ""
+done
+```
+
+And then we can create a machine pool with GPU-enabled instances, in our example i use eu-central-1c AZ:
+
+```bash
+export CLUSTER_NAME=mycluster
+rosa create machine-pool -c $CLUSTER_NAME --name gpu --replicas=1 --availability-zone eu-central-1c --instance-type g4dn.xlarge --use-spot-instances
+```
+
+This command creates a machine pool named "gpu" with one replica using the g4dn.xlarge spot instance, which is x86_64 instance with Nvidia T4 16GB GPU.
+
+## Deploy Required Operators
+
+We'll use kustomize to deploy the necessary operators thanks to this repository provided by Red Hat COP (Community of Practices) [link](https://github.com/redhat-cop/gitops-catalog)
+
+1. Node Feature Discovery (NFD) Operator:
+   ```bash
+   oc apply -k https://github.com/redhat-cop/gitops-catalog/nfd/operator/overlays/stable
+   ```
+   The NFD Operator detects hardware features and configuration in your cluster.
+
+2. GPU Operator:
+   ```bash
+   oc apply -k https://github.com/redhat-cop/gitops-catalog/gpu-operator-certified/operator/overlays/stable
+   ```
+   The GPU Operator manages NVIDIA GPUs drivers in your cluster.
+
+## Create Operator Instances
+
+After the operators are installed, create their instances:
+
+1. NFD Instance:
+   ```bash
+   oc apply -k https://github.com/redhat-cop/gitops-catalog/nfd/instance/overlays/only-nvidia
+   ```
+   This creates an NFD instance for cluster.
+
+2. GPU Operator Instance:
+   ```bash
+   oc apply -k https://github.com/redhat-cop/gitops-catalog/gpu-operator-certified/instance/overlays/aws
+   ```
+   This creates a GPU Operator instance configured for AWS.
+
+## Deploy Ollama and OpenWebUI
+
+Now, let's deploy Ollama for inference and OpenWebUI for interacting with the LLM:
+
+1. Create a new project:
+   ```bash
+   oc new-project llm
+   ```
+
+2. Deploy Ollama:
+   ```bash
+   oc new-app docker.io/ollama/ollama:0.3.10 --import-mode=PreserveOriginal
+   oc patch deployment ollama --type=json -p '[
+     {"op": "remove", "path": "/spec/strategy/rollingUpdate"},
+     {"op": "replace", "path": "/spec/strategy/type", "value": "Recreate"}
+   ]'
+   oc set volume deployment/ollama --add --type=pvc --claim-size=50Gi --mount-path=/.ollama --name=config
+   oc set resources deployment/ollama --limits=nvidia.com/gpu=1
+   ```
+   This deploys Ollama, sets up persistent storage, and allocates a GPU to the deployment.
+
```

Comment:
Add here:

"Use the following command to deploy OpenWebUI deployment, add the necessary storage using the required environment variables"